### DISCUSSION TITLE
force-level1-csd only finds fraction of available images (Landsat-8 & Sentinel-2)

### DISCUSSION THREAD

**Discussion by LuiseMW** [davidfrantz/force#197] [Q&A] [answered]
Hi all,

we are trying to get PC (called PC-1) working with the complete force-workflow and encountered a problem with Sentinel-2 and Landsat-8 image retrieval and download. (force-level1-csd )
When we run the download command for Sentinel or Landsat we only get a fraction of the images available (tested 2017-22, 3 countries, europe and africa). E.g 6 images from FORCE but 120 are available, in ESA Copernicus Scihub / USGS earthexplorer.
We tried .shp, .gpkg and also by tile number for Sentinel.

All other commands run like a charm except for download.
We do not get an Error message but only a few images are found and downloaded.

We do have another PC (called PC-2) running with FORCE and there all functions and download is working as expected.
Runs on Ubuntu 21.10 , 64 cores, RAM 128, GSUTIL 5.6, Python 3.9.7, gdal 3.2.2- installed in Jan'22.
When we run the force-level1-csd on PC-2 we get all images available like in ESA.

First, we had FORCE installed on PC-1 with a ubuntu 18.04 version and then decided to upgrade to a higher version ubuntu 21. as it is running on PC-2. But the problem stayed exactly the same.

PC-1:
1.Installation April '22: ubuntu 18.04  + same specifications as under 2.) 
2.Installation May '22:  ubuntu 21, GDAL 3.2.2, Python 3.9.7 ,GSUTIL 5.10, RAM 60GB, Cores 12,storage multiple add. TB drives, 

Any ideas, what we can additionally try out to solve this?

Best to all,

@ecoinfo-netra
@baldl

**Comment by davidfrantz** [comment]
Hi @LuiseMW,

did you update the metadata catalogue on PC-1?

Cheers,
David

**Comment by LuiseMW** [comment]
Hi @davidfrantz ,

yes we did update the catalog while we had installtion1 and also with installation 2 after the first try-outs again.

Best, Luise

**Comment by davidfrantz** [comment]
hmm, can you please post the output of the tool from both machines when using the ``-n`` option?

and just to be sure, you are running the same FORCE version? ``force -v``

Best,
David

**Reply by LuiseMW** [reply]
PC-1 operates on version 3.7.7 
PC-2 on 3.7.6

force-level1-csd -c 0,100 -n -d 20180101,20180331 -s S2A,S2B ../catalog/ data/force/level1/level1-datapool data/force/level1/level1-datapool/queue.txt data/force/misc/area_extent.shp

Searching for footprints / tiles intersecting with geometries of AOI shapefile...

Querying the metadata catalogue for Sentinel2 data
Sensor(s): S2A,S2B
Tile(s): T37NEH,T37NEJ,T37NFH,T37NFJ
Daterange: 20180101 to 20180331
Cloud cover minimum: 0%, maximum: 100%

/usr/local/bin/force-level1-csd: line 536: [: 18967,139688: integer expression expected
/usr/local/bin/force-level1-csd: line 540: [: 18967,139688: integer expression expected
/usr/local/bin/force-level1-csd: line 544: [: 18967,139688: integer expression expected
35 Sentinel2 Level 1 scenes matching criteria found
18967,1MB data volume found.

Done.


force-level1-csd -n -d 20150101,20191231 -s LC08 ../catalog/ data/force/level1/level1-datapool/Landsat data/force/level1/level1-datapool/Landsat/queue_ls.txt data/force/misc/area_extent.shp

Searching for footprints / tiles intersecting with geometries of AOI shapefile...

Querying the metadata catalogue for Landsat data
Sensor(s): LC08
Tier(s): T1
Tile(s): 167055,167056,168055,168056
Daterange: 20150101 to 20191231
Cloud cover minimum: 0%, maximum: 100%

/usr/local/bin/force-level1-csd: line 536: [: 62164,242191: integer expression expected
/usr/local/bin/force-level1-csd: line 540: [: 62164,242191: integer expression expected
/usr/local/bin/force-level1-csd: line 544: [: 62164,242191: integer expression expected
62 Landsat Level 1 scenes matching criteria found
62164,2MB data volume found.

Done.

**Reply by Baldl** [reply]
PC-2:
 force-level1-csd -c 0,100 -n -d 20180101,20180331 -s S2A,S2B data/force/level1/meta data/force/level1/level1-datapool data/force/level1/level1-datapool/queue.txt data/force/misc/area_extent.shp

Searching for footprints / tiles intersecting with geometries of AOI shapefile...

Querying the metadata catalogue for Sentinel2 data
Sensor(s): S2A,S2B
Tile(s): T37NEH,T37NEJ,T37NFH,T37NFJ
Daterange: 20180101 to 20180331
Cloud cover minimum: 0%, maximum: 100%

105 Sentinel2 Level 1 scenes matching criteria found
60.21GB data volume found.

Done.



force-level1-csd -n -d 20150101,20191231 -s LC08 data/force/level1/meta data/force/level1/level1-datapool/Landsat data/force/level1/level1-datapool/Landsat/queue_ls.txt data/force/misc/area_extent.shp

Searching for footprints / tiles intersecting with geometries of AOI shapefile...

Querying the metadata catalogue for Landsat data
Sensor(s): LC08
Tier(s): T1
Tile(s): 167055,167056,168055,168056
Daterange: 20150101 to 20191231
Cloud cover minimum: 0%, maximum: 100%

328 Landsat Level 1 scenes matching criteria found
338.91GB data volume found.

Done.

**Comment by davidfrantz** [comment]
It looks like PC-1 is running an older FORCE version, which throws some errors. There has been a fix in CSD between those versions. Can you please update and try again?

**Reply by Baldl** [reply]
@davidfrantz thanks for your help, just to make sure: the PC-1, that throws the errors is on version 3.7.7, shouldn´t this be the latest one? The older one (3.7.6) works fine for us...

Best,
Lisa

**Comment by ernstste** [comment, marked as answer]
Hi Luise and Lisa,

I'm unable to reproduce the error on v3.7.7 when querying the same tiles. 

Do both your systems use the same locale?

Stefan

**Reply by Baldl** [reply]
Hi Stefan,

thank you for the hint, they actually do not:

PC-1: runs
$ locale
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=de_DE.UTF-8
LC_TIME=de_DE.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=de_DE.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=de_DE.UTF-8
LC_NAME=de_DE.UTF-8
LC_ADDRESS=de_DE.UTF-8
LC_TELEPHONE=de_DE.UTF-8
LC_MEASUREMENT=de_DE.UTF-8
LC_IDENTIFICATION=de_DE.UTF-8
LC_ALL=

PC-2:
$ locale
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=

Best,
Lisa

**Reply by ernstste** [reply]
Thanks Lisa.

It *should* not make a difference, at least I wasn't able to find problems using the de locale. However, something seems to be off when subsetting the metadata.
Not easy to debug since I can't reproduce the error here. 
The two things I can recommend trying at this point are 
a) redownload the metadata catalogues
b) try running FORCE in docker (this is actually recommended over installing if you have access to docker)

Stefan

**Reply by ecoinfo-netra** [reply]
Dear Stefan and David ,

We installed the docker as suggested by you and it is working now . 
Thank you so much for the help.
- 
@LuiseMW @Baldl

**Reply by ernstste** [reply]
Sorry we couldn't get to the root cause of the issue, glad to hear you got it solved!

Stefan
